src/psyclone/transformations.py

```diff
@@ -2126,8 +2126,80 @@ def apply(self, outer):  # pylint: disable=arguments-differ
         return schedule, keep
 
 
-class ProfileRegionTrans(RegionTrans):
+class OCLTrans(Transformation):
+    '''
+    Switches on/off the generation of an OpenCL PSy layer for a given
+    Schedule. For example:
+
+    >>> invoke = ...
+    >>> schedule = invoke.schedule
+    >>>
+    >>> ocl_trans = OCLTrans()
+    >>> new_sched, _ = ocl_trans.apply(schedule)
+
+    '''
+    @property
+    def name(self):
+        '''Returns the name of this transformation as a string.'''
+        return "OCLTrans"
+
+    def apply(self, sched, opencl=True):
+        '''
+        Apply the OpenCL transformation to the supplied Schedule. This
+        causes PSyclone to generate an OpenCL version of the corresponding
+        PSy-layer routine. The generated code makes use of the FortCL
+        library (https://github.com/stfc/FortCL) in order to manage the
+        OpenCL device directly from Fortran.
+
+        :param sched: Schedule to transform.
+        :type sched: :py:class:`psyclone.psyGen.Schedule`
+        :param bool opencl: whether or not to enable OpenCL generation.
 
+        '''
+        if opencl:
+            self._validate(sched)
+        # create a memento of the schedule and the proposed transformation
+        from psyclone.undoredo import Memento
+        keep = Memento(sched, self, [sched])
```

Comment:
You'll need to keep the value of openCL in the Memento too by adding it to the list.

Reply:
Oh yes, done.